repositories/client/postgresql/pgxpool.go

```diff
@@ -0,0 +1,90 @@
+package postgresql
+
+import (
+	"context"
+	"example/test-items-service/properties"
+	"example/test-items-service/repositories/client"
+	"fmt"
+	"github.com/jackc/pgx/v5/pgxpool"
+	"log"
+	"time"
+)
+
+const (
+	MaxAttempts     = 5
+	AttemptDuration = 5 * time.Second
+)
+
+func GetInstance(context context.Context) (pool client.Client) {
+	log.Println("Creating DB connection pool ...")
+	var err error
+	props := properties.GetProperties()
+	config := newPgConfig(props)
+	pool, err = getClient(context, MaxAttempts, AttemptDuration, config)
+	if pool == nil || err != nil {
+		log.Printf("Unable to create DB client: %v\n", err)
+	}
+	return pool
+}
+
+type pgConfig struct {
+	Username string
+	Password string
+	Host     string
+	Port     string
+	Database string
+}
+
+func newPgConfig(properties *properties.Properties) *pgConfig {
+	return &pgConfig{
+		Username: properties.DB.Username,
+		Password: properties.DB.Password,
+		Host:     properties.DB.Host,
+		Port:     properties.DB.Port,
+		Database: properties.DB.Database,
+	}
+}
+
+func getClient(ctx context.Context, maxAttempts int, maxDelay time.Duration, cfg *pgConfig) (pool client.Client, err error) {
+	dsn := fmt.Sprintf("postgresql://%s:%s@%s:%s/%s",
+		cfg.Username, cfg.Password, cfg.Host, cfg.Port, cfg.Database,
+	)
+
+	err = doWithAttempts(func() error {
```

Comment:
Create separate func instead of anonymous func

Reply:
it's a callback function, I don't know how it will be looks like here in other way. Please provide a solution)